pkg/sysfs/parsers.go

```diff
@@ -115,6 +133,65 @@ func parseNumeric(path, value string, ptr interface{}) error {
 	return err
 }
 
+// parseNumericMap parses a numeric string into integer of the right size.
+func parseNumericMap(path, key, value string, vmap interface{}) error {
+	var numstr string
+	var num, unit int64
+	var unum uint64
+	var f float64
+	var err error
+
+	if numstr, unit, err = splitNumericAndUnit(path, value); err != nil {
+		return err
+	}
+
+	switch vmap.(type) {
+	case map[string]int:
+		num, err = strconv.ParseInt(numstr, 0, strconv.IntSize)
+		vmap.(map[string]int)[key] = int(num * unit)
+	case map[string]int8:
+		num, err = strconv.ParseInt(numstr, 0, 8)
+		vmap.(map[string]int8)[key] = int8(num * unit)
+	case map[string]int16:
+		num, err = strconv.ParseInt(numstr, 0, 16)
+		vmap.(map[string]int16)[key] = int16(num * unit)
+	case map[string]int32:
+		num, err = strconv.ParseInt(numstr, 0, 32)
+		vmap.(map[string]int32)[key] = int32(num * unit)
+	case map[string]int64:
+		num, err = strconv.ParseInt(numstr, 0, 64)
+		vmap.(map[string]int64)[key] = int64(num * unit)
+
+	case map[string]uint:
+		unum, err = strconv.ParseUint(numstr, 0, strconv.IntSize)
+		vmap.(map[string]uint)[key] = uint(unum * uint64(unit))
+	case map[string]uint8:
+		unum, err = strconv.ParseUint(numstr, 0, 8)
+		vmap.(map[string]uint8)[key] = uint8(unum * uint64(unit))
+	case map[string]uint16:
+		unum, err = strconv.ParseUint(numstr, 0, 16)
+		vmap.(map[string]uint16)[key] = uint16(unum * uint64(unit))
+	case map[string]uint32:
+		unum, err = strconv.ParseUint(numstr, 0, 32)
+		vmap.(map[string]uint32)[key] = uint32(unum * uint64(unit))
```

Comment:
Why would we want anything other than `{int64,uint64,float64}` as internal result values? When I see code such as `uint8(unum * uint64(unit))` I start thinking that this is going overflow.

Reply:
Valid point. Updated to use highest-precision internal representations and added checks for trivial overflow/underflow of the numeric part without the unit.